### ISSUE TITLE
Future ideas

### ISSUE THREAD

**Issue by rogeliog** [jest-community/jest-watch-typeahead#9]
Context: It seems that at some point, we should add a way to configure watch plugins. (still need to create an issue for that)

example:
```json
"watchPlugins": [
  ["jest-watch-typeahead/filename", {
    "key": "m",
    "otherOption": true
  }
]
```

I think we can leverage a lot of what we have here to create a robust typeahead solution. It would be awesome to support advance use cases, for example:
- Multiple `jest-watch-typeahad` pugins at the same time
- Allow for custom filers strategies like: `glob`, `regex`, `startsWith`, `exclude`, `etc`.
 
_NOTE: This assume that plugins are "configurable"_
```
  "watchPlugins": [
    "jest-watch-typeahead/testname",
    "jest-watch-typeahead/filename",
    ["jest-watch-typeahead",
      {
        "key": "x",
        "type": "filename",
        "prompt": "exclude file globs",
        "strategy": "exclude-glob"
      }
    ],
    ["jest-watch-typeahead",
      {
        "key": "m",
        "type": "testname",
        "prompt": "run typeahead powered by `my-filter`",
        "strategy": "./my-filter.js"
      }
    ],
  ]
```
This would add 4 plugins
```
> Press p to filter by filename...
> Press t to filter by test name...
> Press x to exclude file globs...
> Press m to run typeahead powered by `my-filter`...
```

**Comment by SimenB** [comment]
I think this is an awesome idea! Could be used to e.g. apply a global filter for just e2e tests, then filter them without getting all other tests.

We should definitely land config (for runners too) in Jest to unblock stuff like this.

**Comment by rogeliog** [comment]
Yes, I agree that runners also really need config

**Comment by rogeliog** [comment]
Closing in favor of #15 and #16 now that configs are released in `facebook/jest`
